src/app/beer_garden/db/mongo/models.py

```diff
@@ -755,8 +757,60 @@ class Garden(MongoModel, Document):
     }
 
     def deep_save(self):
-        for system in self.systems:
-            system.save()
+        logger = logging.getLogger(self.__class__.__name__)
+
+        # if the call to this method is on a child garden object, we ensure that
+        # when saving the systems, unknowns are deleted
+        if self.connection_type != "LOCAL":
+            # import moved here to avoid a circular import loop
```

Comment:
It feels appropriate to me to have this logic contained in its own function.  Then the `deep_save` would just look something like:

```python
def deep_save(self)
    if self.connection_type != "LOCAL":
        self._update_associated_systems()
        
    self.save()
```

Reply:
This is a pattern I see in your code a lot. I'm not against the habit, so sure. Done.